cranelift/filetests/filetests/isa/x64/fcvt.clif

```diff
@@ -200,3 +200,276 @@ block0(v0: i32x4):
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float32_to_sint32_seq %xmm0, %eax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f22(f32) -> i64 {
+block0(v0: f32):
+  v1 = fcvt_to_sint.i64 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float32_to_sint64_seq %xmm0, %rax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f23(f64) -> i32 {
+block0(v0: f64):
+  v1 = fcvt_to_sint.i32 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float64_to_sint32_seq %xmm0, %eax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f24(f64) -> i64 {
+block0(v0: f64):
+  v1 = fcvt_to_sint.i64 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float64_to_sint64_seq %xmm0, %rax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f25(f32) -> i32 {
+block0(v0: f32):
+  v1 = fcvt_to_sint_sat.i32 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float32_to_sint32_sat_seq %xmm0, %eax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f26(f32) -> i64 {
+block0(v0: f32):
+  v1 = fcvt_to_sint_sat.i64 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float32_to_sint64_sat_seq %xmm0, %rax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f27(f64) -> i32 {
+block0(v0: f64):
+  v1 = fcvt_to_sint_sat.i32 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float64_to_sint32_sat_seq %xmm0, %eax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f28(f64) -> i64 {
+block0(v0: f64):
+  v1 = fcvt_to_sint_sat.i64 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   cvt_float64_to_sint64_sat_seq %xmm0, %rax, %r10, %xmm6
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f29(f32x4) -> i32x4 {
+block0(v0: f32x4):
+  v1 = fcvt_to_uint_sat.i32x4 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   movdqa  %xmm0, %xmm6
+;   pxor    %xmm6, %xmm0, %xmm6
+;   maxps   %xmm0, %xmm6, %xmm0
+;   pcmpeqd %xmm6, %xmm6, %xmm6
+;   psrld   %xmm6, $1, %xmm6
+;   cvtdq2ps %xmm6, %xmm1
+;   cvttps2dq %xmm0, %xmm13
+;   subps   %xmm0, %xmm1, %xmm0
+;   cmpps   $2, %xmm1, %xmm0, %xmm1
+;   cvttps2dq %xmm0, %xmm0
+;   pxor    %xmm0, %xmm1, %xmm0
+;   pxor    %xmm1, %xmm1, %xmm1
+;   pmaxsd  %xmm0, %xmm1, %xmm0
+;   paddd   %xmm0, %xmm13, %xmm0
+;   movq    %rbp, %rsp
+;   popq    %rbp
+;   ret
+
+function %f30(f32x4) -> i32x4 {
+block0(v0: f32x4):
+  v1 = fcvt_to_sint_sat.i32x4 v0
+  return v1
+}
+
+;   pushq   %rbp
+;   movq    %rsp, %rbp
+; block0:
+;   movdqa  %xmm0, %xmm7
+;   cmpps   $0, %xmm7, %xmm7, %xmm7
+;   andps   %xmm0, %xmm7, %xmm0
+;   pxor    %xmm7, %xmm0, %xmm7
+;   cvttps2dq %xmm0, %xmm11
+;   movdqa  %xmm11, %xmm0
+;   pand    %xmm0, %xmm7, %xmm0
```

Comment:
The move on line 468 seems unnecessary, as line 467 could be `cvttps2dq %xmm0, %xmm0` instead.

Reply:
This move was indeed unnecesary, and rebasing on #4709 to fix the interaction between the ISLE `xmm_rm_r` constructor and `produces_const` removed it.

Reply:
We might be able to remove this in the future, but for now the additional move doesn't seem too costly.